src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditService.kt

```diff
@@ -269,14 +275,29 @@ class AuditService(
     licence: Licence,
     summary: String,
     changes: Map<String, Any>,
+    userDetails: Map<String, String>? = null,
   ): AuditEvent {
     val authUsername = SecurityContextHolder.getContext().authentication.name
-    val currentUser = staffRepository.findByUsernameIgnoreCase(authUsername)
+    val currentUser = if (userDetails != null) {
+      mapOf(
+        "username" to authUsername,
+        "firstName" to userDetails["forename"],
+        "lastName" to userDetails["surname"],
+      )
+    } else {
+      staffRepository.findByUsernameIgnoreCase(authUsername)?.let {
```

Comment:
this will only return COMs.. wondering if we need to pass the user details any more now we know the user will always exist in the DB.. I guess we didn't at the start of this ticket but do now?

Reply:
The `else` branch of this should never be hit for initial appointment updates, as the `userDetails` should always be populated by having the `notBlank` constraint on the relevant requests.
The `else` branch exists for all other types of audits, as `createAuditEvent` is widely used. When we get the DB set up to house both prison and probation staff, we can likely revert much of this to just check our own DB rather than passing details through